salt/beacons/inotify.py

```diff
@@ -249,7 +249,7 @@ def beacon(config):
                                 if re.search(exclude.keys()[0], event.pathname):
                                     _append = False
                             except Exception:
-                                log.warning('Failed to compile regex: {0}'.format(exclude.keys()[0]))
+                                log.warning('Failed to compile regex: %s', exclude.keys()[0])
```

Comment:
This should be `list(exclude)[0]` for Python 3 compatibility, since `dict.keys()` is not indexable on Python 3. Also, since running `list()` on a dict just gives you a list of the dict's keys, there's no reason to invoke `.keys()`, since this would result in two lists being generated on Python 2. PY2's `.keys()` already allocates a list, and then we're wrapping it in `list()` for Python 3 compatibility already. Using `list(exclude)[0]` will allocate a single list on both Python 2 and Python 3, since it just iterates over the keys and stores them in the list being created.